src/main/java/com/arpnetworking/metrics/mad/parsers/JsonToRecordParser.java

```diff
@@ -586,17 +599,51 @@ private JsonToRecordParser(final Builder builder) {
         }
     };
 
-    private static final String PREFIXED_HOST_KEY = "_host";
-    private static final String PREFIXED_SERVICE_KEY = "_service";
-    private static final String PREFIXED_CLUSTER_KEY = "_cluster";
+    private static final Function<Version2g.Sample, Quantity> VERSION_2G_SAMPLE_TO_QUANTITY = sample -> {
+        if (sample != null) {
+            if (Double.isFinite(sample.getValue())) {
+                final Unit sampleUnit = sample.getUnit2g() != null
+                        ? CompoundUnit.getExistingUnit(Iterables.getFirst(sample.getUnit2g().getNumerators(), null))
```

Comment:
@vjkoskela "I'll have to take a closer look at this, but I want to stress that we have a good separation between the serialized types and the internal ones. This class is the mapping layer."